lib/iris/coords.py

```diff
@@ -1536,10 +1617,13 @@ def __eq__(self, other):
 
     # The __ne__ operator from Coord implements the not __eq__ method.
 
-    # This is necessary for merging, but probably shouldn't be used otherwise.
-    # See #962 and #1772.
-    def __hash__(self):
-        return hash(id(self))
+    # For Python 3, we must explicitly re-implement the '__hash__' method, as
+    # defining an '__eq__' has blocked its inheritance.  See ...
+    # https://docs.python.org/3.1/reference/datamodel.html#object.__hash__
+    # "If a class that overrides __eq__() needs to retain the
+    # implementation of __hash__() from a parent class, the interpreter
+    # must be told this explicitly".
+    __hash__ = Coord.__hash__
```

Comment:
I could use "super" here, if you think it would be preferable (it avoids saying who your parent is).
This way is more efficient though :wink: